### PR TITLE
[consolidated-events] Add types for consolidated-events

### PR DESCRIPTION
**Pull request by romellem**
Adding types for https://www.npmjs.com/package/consolidated-events

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/consolidated-events/.eslintrc.json
Comment (peterblazejewicz) [inline types/consolidated-events/.eslintrc.json:1]: why do you need this file?

**Review by gabritto — CHANGES_REQUESTED** [review]
Please remove the `.eslintrc.json` empty file, and I'll approve this.

**Comment by romellem** [comment]
Ah, in the [CI error](https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/8591863153/job/23541182815), it says

> Error: Should not contain 'tslint.json'. This file is no longer required; place all lint-related options into .eslintrc.json.

And I took that to mean I needed the file; just misread the error.

Thanks!

**Review by gabritto — APPROVED** [review]

**Comment by romellem** [comment]
Ready to merge

**Pull request merged**